dm/syncer/data_validator.go

```diff
@@ -198,3 +402,158 @@ func (v *DataValidator) Stage() pb.Stage {
 	defer v.RUnlock()
 	return v.stage
 }
+
+func (v *DataValidator) startValidateWorkers() {
+}
+
+func (v *DataValidator) dispatchRowChange(key string, row *rowChange) {
+}
+
+func (v *DataValidator) processRowsEvent(header *replication.EventHeader, ev *replication.RowsEvent) error {
+	sourceTable := &filter.Table{
+		Schema: string(ev.Table.Schema),
+		Name:   string(ev.Table.Table),
+	}
+	fullTableName := sourceTable.String()
+	if _, ok := v.unsupportedTable[fullTableName]; ok {
+		return nil
+	}
+
+	targetTable := v.syncer.route(sourceTable)
+	tableInfo, err := v.syncer.schemaTracker.GetTableInfo(sourceTable)
+	if err != nil {
+		if schema.IsTableNotExists(err) {
+			// not a table need to sync
+			return nil
+		}
+		return terror.Annotate(err, "failed to get table info")
+	}
+
+	if len(tableInfo.Columns) < int(ev.ColumnCount) {
+		v.unsupportedTable[fullTableName] = "binlog has more columns than current table"
+		return nil
+	}
+
+	var primaryIdx *model.IndexInfo
+	for _, idx := range tableInfo.Indices {
+		if idx.Primary {
+			primaryIdx = idx
+		}
+	}
+	if primaryIdx == nil {
```

Comment:
there will be no PK in tableInfo.Indices when PKIsHandle in my memory. We might provide a function for this case as well as for `handlePkExCase`

Reply:
`PKIsHandle = true` when table has primary key like `id int primary key CLUSTERED`, since schema-tracker(we get from it) only use downstream DDL when the task is incremental only, will support this case later.